src/parser/html.py

```diff
@@ -0,0 +1,35 @@
+from src.reader.reader import Reader
+from bs4 import BeautifulSoup
+
+TAGS = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'li']
+
+class HTMLParser():
+    """
+    HTMLParser parses text content from HTML files using the provided reader.
+    """
+    def __init__(self, reader):
+        if not isinstance(reader, Reader):
+            raise ValueError("HTMLParser requires a Reader object.")
+        self.reader = reader
+
+    def parse(self):
+        """
+        Parse the next HTML file from the reader and return the entire text.
+        """
+        try:
+            html = self.reader.read()
+        except Exception as e:
+            raise ValueError("Error retrieving file from reader: " + str(e))
+        if html is None:
+            return None
+        return self._parse_text(html)
+
+    def _parse_text(self, html):
+        """
+        Parse the text from the provided HTML.
+        """
+        soup = BeautifulSoup(html, 'html.parser')
+        text = ""
+        for tag in soup.find_all(TAGS):
+            text += tag.text + " "
+        return text.strip()
```

Comment:
In the future, we may want to be able to seek to different parts of the text that are more vocabulary/topic-dense (e.g focusing more on the title vs the last paragraph). I think it's ok that we're kind of leveling the playing field and treating all tags from the text equally for now, but it might be something we want to revisit one day.

Reply:
That's a great point, for topic modeling the headline seems way more important, or maybe the only important part of the article.